File: Python/FlatMap.py
arr = [1, [2, 3], [4, [5, [6, 7], 8]], 9, [10]]


def flat_map(array):
    res = []
    for item in array:
        if type(item) == list:
            res = [*res, *flat_map(item)] # unpacking list with *
        else:
            res.append(item)
    return res


print(flat_map(arr))

File: JavaScript/FlatMap.js
const arr = [1, [2, 3], [4, [5, [6, 7], 8]], 9, [10]];

Array.prototype.flattenMap = function () {
  let res = [];
  this.forEach((item, _index) => {
    if (Array.isArray(item)) {
      res = [...res, ...this[_index].flattenMap()];
    } else res.push(item);
  });
  return res;
};

console.log(arr.flattenMap());
